src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -622,8 +625,19 @@ private void RemoveEventHandlerIds(ArrayRange<int> eventHandlerIds, Task afterTa
                 // any further). We must clone the data because the underlying RenderBatchBuilder
                 // may be reused and hence modified by an unrelated subsequent batch.
                 var eventHandlerIdsClone = eventHandlerIds.Clone();
-                afterTask.ContinueWith(_ =>
-                    RemoveEventHandlerIds(eventHandlerIdsClone, Task.CompletedTask));
+
+                try
+                {
+                    await afterTaskIgnoreErrors;
+                }
+                catch (Exception)
+                {
+                    // As per method contract, we're not error-handling the task.
+                    // That remains the caller's business.
+                }
+
+                // We know the next execution will complete synchronously, so nothing to await
+                _ = RemoveEventHandlerIdsAsync(eventHandlerIdsClone, Task.CompletedTask);
```

Comment:
Is there a good reason to do this vs just extracting this L612-618 to a local function?

Reply:
This logic is how it already is prior to this PR. We could refactor further, but it’s not required for this PR.